### PR TITLE
refactor(ui): reorganize `shared/utils.ts`

### PR DESCRIPTION
**Pull request by agilgur5**
While writing various PRs, I've had some of these changes in a stash for a while. When working on some recent namespace fix PRs, this refactor felt particularly necessary given all the unrelated functions to namespaces in this file

### Motivation

- this file's main purpose is for namespace manipulation, so rename to `shared/namespaces.ts`
  - and rename `Utils` to `nsUtils` for explicitness as well 
  - generally speaking, "dumping grounds" are not good to have and organizing them when they get large is best practice

- these changes also make these functions easier to unit test and give them proper encapsulation
  - private functions are no longer exported

### Modifications

- `shared/utils.ts` -> `shared/namespaces.ts`
  - `Utils` -> `nsUtils`
   
- several util functions were used in exactly one place, so just move them to that place
  - `statusIconClasses` was only used in `phase-icon.tsx`
  - `isWorkflowRunning`, `isWorkflowSuspended` were only used in `workflow-operations-map.tsx`

- some were only used together with exactly one component, so move to that component
  - `getValueFromParameter` was only used with `ParametersInput`

- others were only used for one section of the UI, e.g. Events or Workflows
  - so move them to a `utils.ts` file there; they are not shared between sections
  - similarly `queryParams` was only used in `services/`

- also simplify some of these functions:
  - with optional chaining
  - by inverting unnecessarily nested conditionals
  - by removing unnecessary singleton structures

### Verification

- Purely stylistic changes, no semantic changes
- `yarn lint`, `test` & `build` all pass

### CODE REVIEW

File: ui/src/app/sensors/utils.ts [hunk 1]
```diff
@@ -1,28 +1,22 @@
...
-                    isRunning = true;
-                }
-            });
-            if (hasFailed) {
-                classes = [icon, 'status-icon--failed'];
-            } else if (isRunning) {
-                classes = [icon, 'status-icon--spin'];
-            } else {
-                classes = [icon, 'status-icon--running'];
-            }
```
Comment (agilgur5) [inline ui/src/app/sensors/utils.ts:24]: I simplified this quite a bit as the two flags, `hasFailed` and `isRunning`, are mutually exclusive (by the logic here but also in general), so can just return as soon as one of them is hit

File: ui/src/app/shared/namespaces.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    if (value != null) {
+        localStorage.setItem(currentNamespaceKey, value);
+    } else {
+        localStorage.removeItem(currentNamespaceKey);
+    }
+    onNamespaceChange(getCurrentNamespace());
+}
+
+export function getCurrentNamespace() {
+    return fixLocalStorageString(localStorage.getItem(currentNamespaceKey)) ?? (getUserNamespace() || getManagedNamespace());
```
Comment (agilgur5) [inline ui/src/app/shared/namespaces.ts:53]: ack part of this is accidentally taken from #13347 (merge conflict/stash issues); I'm gonna mark this as draft temporarily as a result. Will rebase once that's merged

Comment (agilgur5) [reply ui/src/app/shared/namespaces.ts:53]: Rebased

**Review by isubasinghe — CHANGES_REQUESTED** [review]
Looks okay, can you please make the CI pass?

**Comment by agilgur5** [comment]
Yea, looks like I accidentally added an extra new line during conflict resolution

**Comment by agilgur5** [comment]
Should pass now

**Review by isubasinghe — APPROVED** [review]
Code looks fine, I am unsure if we should merge this personally though, just due to future potential git conflicts.

**Comment by agilgur5** [comment]
> I am unsure if we should merge this personally though, just due to future potential git conflicts.

Same as I wrote in https://github.com/argoproj/argo-workflows/pull/12539#issuecomment-1901012046, this logic applies to any refactor so this is a strawman, as we ofc don't avoid refactors.
Same as there, conflicts are (much) worse for dep upgrades than code changes as well, but we don't avoid those either.

**Pull request merged**